gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationsResource.java

```diff
@@ -149,18 +164,50 @@ public Response getApplications(
             List<String> idsOrderedBySubscriptions = new ArrayList<>(
                 filteringService.getApplicationsOrderByNumberOfSubscriptions(applicationIds)
             );
-            orderingComparator = Comparator.comparingInt(idsOrderedBySubscriptions::indexOf);
+            orderingComparator = Comparator.comparingInt(value -> idsOrderedBySubscriptions.indexOf(value.getId()));
         } else {
             orderingComparator = Comparator.comparing(ApplicationListItem::getName, String.CASE_INSENSITIVE_ORDER);
         }
+
+        List<ApplicationListItem> applicationsList = appSupplier.get().sorted(orderingComparator).collect(Collectors.toList());
         boolean isAsc = !order.startsWith("-");
         if (!isAsc) {
-            orderingComparator.reversed();
+            Collections.reverse(applicationsList);
         }
+        return createListResponse(applicationsList, paginationParam);
+    }
 
-        List<ApplicationListItem> applicationsList = appSupplier.get().sorted(orderingComparator).collect(Collectors.toList());
+    @Override
+    protected Map<String, Map<String, Object>> computeMetadata(
```

Comment:
`computeMetadata` was currently only technically computing maps of metadata.
I fell like it has been hijacked to do other functional stuff.

Maybe we could find a better pattern for that ? Or at least rename it to better show what it does ?

Reply:
✅